### ISSUE TITLE
Create Unvanquished icon font

### ISSUE THREAD

**Issue by Gireen** [Unvanquished/Unvanquished#3269]
With more icons getting available as vector graphics it becomes possible to create our own icon font to also benefit from this ingame.
In different resolutions and scales icons will still look sharp and beacons tags can be replaced with the actual unit icons. 

Some hurdles are that the process of maintenance changes quiet a bit. 

- It needs some easy available font build tool. There are many online services but something easy to automate would be better. 
- For easy development and not yet available vector graphics its needed to replace font icons with images #3268
- The game needs also support for multiple fonts for icons.

**Comment by illwieckz** [comment]
Good idea, though I have myself no idea about “how to make a font”. Do you have some experience in that?

**Comment by sweet235** [comment]
We could script fontforge to create a font out of svg files. But there is also an SVG plugin for RMLUI, which might be the easier method to load vector graphics.

Font files are notoriously bad for incremental updates.

**Comment by sweet235** [comment]
Notice that we can simply make urcheon convert SVG files to pixel images, solving this problem by a oneliner.

**Comment by Gireen** [comment]
The automatic conversion is not the problem, i think urcheon does that already with all images into crn.  

So the plugin would be
https://mikke89.github.io/RmlUiDoc/pages/cpp_manual/svg.html

And requires [LunaSVG](https://github.com/sammycage/lunasvg) as dependency.

Is there anything blocking or downside to just adding this to Unvanquished?

**Comment by slipher** [comment]
> Is there anything blocking or downside to just adding this to Unvanquished?

A possible downside of SVG is if it results in more polygons being generated, which can hurt performance since they all have to be sent over IPC and uploaded to the GPU. Of course the font renderer we have already works on a similar principle and generates a number of tiny polygons for every letter, but we would need to check that the SVG implementation is not even more costly.

**Comment by DolceTriade** [comment]
RmlUI's implementation of SVG generates a quad with the SVG as a texture. Generates it once per instance of the SVG depending on the size. This means two tris / SVG?

**Comment by slipher** [comment]
> RmlUI's implementation of SVG generates a quad with the SVG as a texture. Generates it once per instance of the SVG depending on the size. This means two tris / SVG?

Cool, sounds appropriate for something used like a font.

**Comment by Gireen** [comment]
A difference i remembered is that font icons allow the use of font effects like outlines.

Is there something similar possible with SVG without editing the image files?
